yabt/buildcontext.py

```diff
@@ -464,11 +464,25 @@ def build(target: Target):
                 if run_tests and 'testable' in target.tags:
                     if self.conf.no_test_cache or not test_cached:
                         logger.info('Testing target {}', target.name)
-                        test_start = time()
-                        self.test_target(target)
+                        retries = target.props.retries or self.conf.test_retries or 3
+                        if retries < 1:
+                            raise ValueError('Retries value must be positive: got {}'.format(retries))
+                        test_start = 0
+                        for fails in range(0, retries + 1):
+                            test_start = time() # TODO(bergden) time all retries?
```

Comment:
@itamaro Next PR should do the accumulator stuff, so at this point I just time the last run, but ideally we would enable gathering statistics on the successful runs (and maybe the failed runs)